initialization/initialization.go

```diff
@@ -170,14 +203,38 @@ func (init *Initializer) Initialize(ctx context.Context) error {
 
 	numLabels := uint64(init.opts.NumUnits) * uint64(init.cfg.LabelsPerUnit)
 	fileNumLabels := numLabels / uint64(init.opts.NumFiles)
+	difficulty := shared.PowDifficulty(numLabels)
 
 	init.logger.Info("initialization: starting to write %v file(s); number of units: %v, number of labels per unit: %v, number of bits per label: %v, datadir: %v",
 		init.opts.NumFiles, init.opts.NumUnits, init.cfg.LabelsPerUnit, init.cfg.BitsPerLabel, init.opts.DataDir)
 
 	for i := 0; i < int(init.opts.NumFiles); i++ {
-		if err := init.initFile(ctx, uint(init.opts.ComputeProviderID), i, numLabels, fileNumLabels); err != nil {
+		if err := init.initFile(ctx, i, numLabels, fileNumLabels, difficulty); err != nil {
+			return err
+		}
+	}
+
+	if init.nonce == nil {
+		// continue searching for a nonce
+		res, err := oracle.WorkOracle(
```

Comment:
A user that finished initialization of their PoST, hasn't found a nonce, stops and restarts will have to start looking for a nonce at the `numLabels` index again.

The only way I see around this is to add the last position that was passed to WorkOracle to the metadata file. This would require a few additional checks (and tests for them) as well:

- what if `lastPosition` is less than `numLabels` and post file size doesn't match? - ignore `lastPosition` and continue initialization as it is implemented at the moment?
- what if `lastPosition` is greater than `numLabels` but post file size doesn't match? - restart?

Reply:
`initFile` already supports the initialization recovery by reading the number of written bytes on disk (in contrary to the metadata file, which is less reliable).
You only need to take care of the PoW-only mode recovery, by persisting `lastPosition` in the metadata file prior to every work oracle invocation at that stage (and not during initialization).

> - what if `lastPosition` is less than `numLabels` and post file size doesn't match? - ignore `lastPosition` and continue initialization as it is implemented at the moment?

This can happen if the user increased the number of units (or due to a bug or disk corruption). You should ignore `lastPosition` and continue initialization as-is. 

> - what if `lastPosition` is greater than `numLabels` but post file size doesn't match? - restart?

Same, if the user decreased the number of units. You should ignore `lastPosition` and continue initialization as-is.

Reply:
I added `LastPosition` to `Metadata`. It is set when no nonce is found below `numLabels` and if stopped and continued `Initializer` will start looking for nonces at higher positions.